Lib/test/test_descr.py

```diff
@@ -1835,29 +1835,24 @@ def __init__(self, foo):
         object.__init__(A(3))
         self.assertRaises(TypeError, object.__init__, A(3), 5)
 
-    @unittest.expectedFailure
     def test_restored_object_new(self):
         class A(object):
             def __new__(cls, *args, **kwargs):
-                raise AssertionError
-        self.assertRaises(AssertionError, A)
+                raise TypeError('A')
+        self.assertRaisesRegex(TypeError, 'A', A)
         class B(A):
             __new__ = object.__new__
-            def __init__(self, foo):
-                self.foo = foo
-        with warnings.catch_warnings():
-            warnings.simplefilter('error', DeprecationWarning)
-            b = B(3)
-        self.assertEqual(b.foo, 3)
-        self.assertEqual(b.__class__, B)
+            def __init__(self):
```

Comment:
Ok, I see. This needs to be allowed.

Reply:
The point of the test is that if `__init__` have 1 parameter and `__new__` accept arbitrary arguments, it should be possible to call the constructor with 1 argument.

Reply:
Yes, I followed the issue you've linked. Now I understand that `__new__ = object.__new__` should not matter 👍